draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -683,6 +821,97 @@ Elligator2 can be simplified with projective coordinates.
 
 ((TODO: write this variant))
 
+## FT-hash
+
+{{FT12}} outline a method for hashing onto a set of curves which includes BN curves, which are suitable
+for bilinear pairings. A number of cryptographic schemes rely on pairings, including some identity-based
+encryption (IBE) and attribute-based encryption (ABE) schemes.  BN curves are not one specific curve but a
+class of curves which can be constructed using Barreto and Naehrig's algorithm {{BN05}}. BN curves are a
+preferred option for asymmetric pairings as they provide relatively efficient computation for pairings, which are
+usually the slowest computation in any cryptosystem using them. There is also work giving parameter selection
+for 256-bit BN-curves, resulting in 128 bits of security for hashing onto BN curves. Suggestions for suitable 224,
+256, 384, and 512-bit curves which are compliant with the ISO document and are based on M-type twisted curve can
+be found in {{ISO09EC}}. Other work on parameter selection for suitable curves can be found in {{BGMO+10}}
+{{AKLG+11}} {{Scott13}} {{BCMN+16}}.
+
+This method is suitable for BN-curves, which have the form E: f(x) = y^2 = x^3 + b, where operations
+are over a field Fp,  where p = 7 (mod 12) and 1 + b is a nonzero square in Fp. Fouque and Tibouchi give an 
+encoding F which makes the hash function Hash(m) = F(H1(m)) + F(H2(m)) indifferentiable from a random oracle,
+where  H1 and H2 are independent hash functions into Fp. We give this encoding and hash function construction
+in the steps below.
+
+~~~
+FT-encoding(t)
```

Comment:
This should be rewritten as map2curve_BN(t), as this is a generic encoding mechanism for BN curves.